src/compiler/program.ts

```diff
@@ -991,7 +991,13 @@ namespace ts {
         return sortAndDeduplicateDiagnostics(diagnostics);
     }
 
-    export function formatDiagnostics(diagnostics: Diagnostic[], host: CompilerHost): string {
+    export interface FormatDiagnosticsHost {

```

Comment:
Thanks @weswigham!

Although that sort of rephrases "Format Diagnostics Host" rather than answers any of the questions above ;-)